### ISSUE TITLE
Login system for mainland China (WeChat?)

### ISSUE THREAD

**Issue by riipah** [VocaDB/vocadb#346]
Chinese users are not able to register on VocaDB due to reCAPTCHA restriction. reCAPTCHA is not available in mainland China. As an alternative, a SSO login by Chinese provider could be used.

WeChat provides SSO authentication similar to Twitter. [There is documentation in English.](http://open.wechat.com/cgi-bin/newreadtemplate?t=overseas_open/docs/web/login/login).

**Comment by riipah** [comment]
Registering application with Weibo apparently requires a business account, and the registration is in Chinese only, so no luck there. Others are saying Weibo would be a better option anyway. The application registration process for Weibo is in Chinese as well, but there are [some instructions](https://auth0.com/docs/connections/social/weibo) in English.

**Comment by AgFlore** [comment]
> Chinese users are not able to register on VocaDB due to reCAPTCHA restriction. reCAPTCHA is not available in mainland China. As an alternative, a SSO login by Chinese provider could be used.

......Well, the easiest and most straightforward alternative would be simply replacing all `https://www.google.com/recaptcha` with `https://www.recaptcha.net/recaptcha`.

Complexity-trivial, right?

**Comment by riipah** [comment]
That URL https://www.recaptcha.net/recaptcha is giving me 404

**Comment by AgFlore** [comment]
> That URL https://www.recaptcha.net/recaptcha is giving me 404

Yes but endpoints like https://www.recaptcha.net/recaptcha/api.js and https://www.recaptcha.net/recaptcha/api/siteverify are functional.

**Comment by riipah** [comment]
It doesn't need API key?

**Comment by AgFlore** [comment]
> It doesn't need API key?

Sure it does need. By my understanding it's an alternative alias of the google.com service.

Document [reads](https://developers.google.com/recaptcha/docs/faq#does-recaptcha-use-cookies)
> reCAPTCHA sets a necessary cookie (_GRECAPTCHA) when executed for the purpose of providing its risk analysis. If you prefer to not use the www.google.com domain which may have other cookies set, you can use www.recaptcha.net instead.

**Comment by riipah** [comment]
I see. So it's the same service, the same API key should work and it's still provided by Google, but it's not blocked in China?

**Comment by AgFlore** [comment]
> I see. So it's the same service, the same API key should work and it's still provided by Google, but it's not blocked in China?

Supposed to be so. You see the endpoint is https-certified by Google Trust Services. It's just somehow poorly decorated, and running without welcome pages. Maybe we can tempoarily enable registering on beta site and test its functionality.

**Comment by ycanardeau** [comment]
I can change the endpoint on the beta site, but I'd say it's only a matter of time before the domain gets blocked.

Reference(s):
https://stackoverflow.com/questions/57827914/google-recaptcha-in-china

**Comment by AgFlore** [comment]
> I can change the endpoint on the beta site, but I'd say it's only a matter of time before the domain gets blocked.
> Reference(s): https://stackoverflow.com/questions/57827914/google-recaptcha-in-china

Not really. [Geoping](http://ping.chinaz.com/www.recaptcha.net) it and you will see, requests for www.recaptcha.net from inside the PRC are actually served by Google's local datacenters inside PRC (just like in other regions), and there's no across-the-wall traffic for users. There were reports of IaaSs blocking access to the domain (likely because they want to market their own captcha solution), but for most users this service is well accessible. 

Therefore, so long as the building of Google still stands there in China, the service won't be "blocked"; and so long as they manage to keep the power switch on (without being cornered to turn off by the authorities, like what happened to the search engine ten years ago), the service won't be down.

Speaking of the current situation, Google China is still running ads.google.cn, translate.google.cn, tensorflow.google.cn, golang.google.cn and a number of other services, and there's no signal of Google's quitting from providing these services in China now.
